tests/test_tacacs_server_group.rb

```diff
@@ -17,98 +14,447 @@
+
+    aaa_group2 = TacacsServerGroup.new(group_name2)
+
+    aaa_group3 = TacacsServerGroup.new(group_name3)
+
+    groups = TacacsServerGroup.groups
+    refute_empty(groups, 'Error: TacacsServerGroup collection is not filled')
+    assert_equal(3, groups.size,
+                 'Error: TacacsServerGroup collection reporting incorrect size')
+    assert(groups.key?(group_name1),
+           "Error: TacacsServerGroup collection does contain #{group_name1}")
+    assert(groups.key?(group_name2),
+           "Error: TacacsServerGroup collection does contain #{group_name2}")
+    assert(groups.key?(group_name3),
+           "Error: TacacsServerGroup collection does contain #{group_name3}")
+
+    detach_aaaservergroup(aaa_group1)
+    detach_aaaservergroup(aaa_group2)
+    detach_aaaservergroup(aaa_group3)
+  end
+
+  def test_servers_tacacs
+    clean_tacacs_config
+    server_name1 = 'server1'
+    server_name2 = 'server2'
+    server1 = create_tacacsserverhost(server_name1)
+    server2 = create_tacacsserverhost(server_name2)
+
+    aaa_group = TacacsServerGroup.new('Group1')
+
+    # pre check that default servers are empty
+    default_server = aaa_group.default_servers
+    assert_empty(default_server, 'Error: Default Servers are not empty')
+
+    aaa_group.servers = [server_name1, server_name2]
+
+    # Check collection size
+    servers = aaa_group.servers.keys
+    assert_equal(2, servers.size,
+                 'Error: Collection is not two servers')
+    assert(servers.include?('server1'),
+           "Error: Collection does not contain #{server_name1}")
+    assert(servers.include?('server2'),
+           "Error: Collection does not contain #{server_name2}")
+
+    detach_aaaservergroup(aaa_group)
+    detach_tacacsserverhost(server1)
+    detach_tacacsserverhost(server2)
+  end
+
+  def test_add_server_tacacs
+    server_name1 = 'server1'
+    server_name2 = 'server2'
+    server1 = create_tacacsserverhost(server_name1)
+    server2 = create_tacacsserverhost(server_name2)
+
+    aaa_group = TacacsServerGroup.new('Group1')
+    aaa_group.servers = [server_name1, server_name2]
+
+    assert_show_match(command: 'show run tacacs+ all | no-more',
+                      pattern: /server server1/)
+    assert_show_match(command: 'show run tacacs+ all | no-more',
+                      pattern: /server server2/)
+
+    detach_aaaservergroup(aaa_group)
+    detach_tacacsserverhost(server1)
+    detach_tacacsserverhost(server2)
+  end
+
+  def test_add_server_twice_tacacs
+    clean_tacacs_config
+    server_name1 = 'server1'
+    server_name2 = 'server2'
+    server1 = create_tacacsserverhost(server_name1)
+    server2 = create_tacacsserverhost(server_name2)
+
+    aaa_group = TacacsServerGroup.new('Group1')
+    # aaa_group.servers = [server_name1, server_name2, server_name2]
+    # this behavior is different on n9k vs n3k, so comment out for now
+    # n3k throws a CLIError and n9k silently ignores
```

Comment:
If this isn't a valid test then get rid of the comment (my vote).
If it is valid then it sounds like it "works" for n9k -- if it configs name1 & name2 ;
for n3k you can rescue it and validate/skip the result.